docs/new_agents.rst

```diff
@@ -1,3 +1,390 @@
+of this file is as follows (using the Arduino as an example):
+
+::
+
+        # SOCS Arduino Agent
+        # socs Agent container for interacting with an Arduino
+
+        # Use socs base image
+        FROM socs:latest
+
+        # Set the working directory to registry directory
+        WORKDIR /app/agents/arduino/
+
+        # Copy this agent into the app/agents directory
+        COPY . /app/agents/arduino/
+
+        # Run registry on container startup
+        ENTRYPOINT ["python3", "-u", "arduino_agent.py"]
+
+
+In this case, the ``WORKDIR``, ``COPY``, and ``ENTRYPOINT`` arguments are all set 
+specifically to the correct directories and files for the Arduino agent. You can 
+additionally connect the container to a Crossbar (WAMP) server; see the Sisock 
+documentation for more on this. 
+
+To include your new Agent among the services provided in your OCS Docker 
+container, navigate to the ``docker-compose.yml`` file in the same sub-directory 
+as your ``default.yaml`` or ``your_institution.yaml`` file. Within 
+``docker-compose.yml``, you should find (or create) a list of services that the 
+docker container provides. You can add your new agent following the example format:
+
+::
+
+  services:
+    arduino:
+      image: grumpy.physics.yale.edu/sisock-data-feed-server:v0.2.12-1-g52852b4
+      environment:
+          TARGET: arduino
+          NAME: 'arduino'
+          DESCRIPTION: "arduino"
+          FEED: "amplitudes"
+      logging:
+        options:
+          max-size: "20m"
+          max-file: "10"
+
+The ``image`` line of this template corresponds to your computer's live feed
+server, which should be the same for all of your agents. The ``image`` entry
+contains entries that allow the live feed to subscribe to the data
+you are reading (under ``environment``), as well as entries for keeping logs
+of your agent's activity (under ``logging``). The ``environment`` entries are:
+
+- ``TARGET``: the same as the ``instance-id`` that you added in the previous
+file. This is used to identify the agent you wish to monitor.
+- ``NAME``: the name that appears in a live feed field name.
+- ``DESCRIPTION``: a short description of the feed you are subscribing to (can
+be a word or a short sentence).
+- ``FEED``: the type of data you are reading. This must match the data type
+used in the ``self.agent.register_feed()`` entry in your agent class.
+
+The ``logging`` options limit the maximum file size of the logs and
+automatically rotates them. This should generally remain constant for all of
+your agents.
+
+Final Steps
+-----------
+After setting up the agent, you can run it from the command line with
+
+::
+
+        python3 agent_name.py --instance-id=arduino
+
+Here ``--instance-id`` is the same as that given in your ocs-site-configs
+``default.yaml`` file. The agent will then run until it is manually ended. Once
+you have a successfully running Agent, then you can build a Docker image for
+it.
```

Comment:
After containerizing your Agent you no longer need to run it on the commandline. Once added to your `docker-compose.yaml` file it'll startup with the rest of your containers when running `docker-compose up -d`.